### PR TITLE
feat: change task after deadline bb-446

### PR DESCRIPTION
**Pull request by Shcracoziabra**
#446 

### Overview

- [x] Created a modal that will appear on the Tasks screen when the countdown for the exact task ends.
- [x] Changed the status of a task after its deadline after user action (extend deadline, mark as completed or skip).
- [x] Modal appears in the Tasks tab if there is at least one task expired.
- [x] User can horisontally slide expired tasks if there are several ones.
- [x] Notification is triggered on the task dueDate property to warn user of the task expired.

### Summary:

- added @notifee/react-native to display expired task notification;
- created ExpiredTasksModal component for expired tasks;
- adjusted DeadlineCountdown component to change appearance when expired or active;
- adjusted TaskCard component to change appearance when expired or active;
- added updateTaskAfterDeadline method to tasks api;
- created updateTaskAfterDeadline thunk for tasks slice;
- added activeTasks and expiredTasks to the tasks slice to make task screen update on state change possible;
- added setActiveTasks, setExpiredTasks, addExpiredTask to slice actions.

### Screenshots

expire tasks modal
push notifications

### Userflow
<video src="https://github.com/user-attachments/assets/4689b6dc-e665-47d4-a0a8-739c71761b34" />

### CODE REVIEW

File: apps/mobile/src/libs/components/deadline-countdown/libs/hooks/use-countdown.hook.ts [hunk 1]
```diff
@@ -5,7 +5,15 @@ import { type Countdown } from "~/libs/types/types";
...
 
-const useCountdown = (deadline: string): Countdown => {
+type useCountdownArguments = {
+	deadline: string;
+	onExpire?: () => void;
+};
+
+const useCountdown = ({
+	deadline,
+	onExpire = (): void => {},
```
Comment (revenets) [inline apps/mobile/src/libs/components/deadline-countdown/libs/hooks/use-countdown.hook.ts:15]:
```suggestion
	onExpire,
```

File: apps/mobile/src/libs/components/deadline-countdown/libs/hooks/use-countdown.hook.ts [hunk 2]
```diff
@@ -29,6 +37,7 @@ const useCountdown = (deadline: string): Countdown => {
 			setCountdown(updatedCountdown);
 
 			if (isExpired(updatedCountdown)) {
+				onExpire();
```
Comment (revenets) [inline apps/mobile/src/libs/components/deadline-countdown/libs/hooks/use-countdown.hook.ts:40]:
```suggestion
				onExpire?.();
```

File: apps/mobile/src/libs/components/deadline-countdown/deadline-countdown.tsx [hunk 3]
```diff
@@ -1,31 +1,55 @@
...
+	isExpired?: boolean;
+	onExpire?: () => void;
 };
 
-const DeadlineCountdown: React.FC<Properties> = ({ deadline }) => {
-	const { days, hours, minutes } = useCountdown(deadline);
+const DeadlineCountdown: React.FC<Properties> = ({
+	deadline,
+	isExpired = false,
+	onExpire = (): void => {},
```
Comment (revenets) [inline apps/mobile/src/libs/components/deadline-countdown/deadline-countdown.tsx:20]:
```suggestion
	onExpire,
```

File: apps/mobile/src/libs/components/task-card/task-card.tsx [hunk 4]
```diff
@@ -19,30 +19,48 @@ import { styles } from "./styles";
...
+	variant?: "active" | "expired";
 };
 
 const EDIT_ICON_SIZE = 20;
 
-const TaskCard: React.FC<Properties> = ({ onComplete, onSkip, task }) => {
+const TaskCard: React.FC<Properties> = ({
+	onComplete = (): void => {},
+	onExpire = (): void => {},
+	onSkip = (): void => {},
```
Comment (revenets) [inline apps/mobile/src/libs/components/task-card/task-card.tsx:33]:
```suggestion
	onComplete,
	onExpire,
	onSkip,
```

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/expired-tasks-modal/expired-tasks-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+import { styles } from "./styles";
+
+type Properties = {
+	tasks: TaskDto[];
+};
+
+const ExpiredTasksModal: React.FC<Properties> = ({ tasks }) => {
+	const { width } = useWindowDimensions();
+
+	const scrollViewWidth = useMemo(() => tasks.length * width, [tasks, width]);
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/expired-tasks-modal/expired-tasks-modal.tsx:30]:
```suggestion
	const scrollViewWidth = useMemo(() => {
	  return tasks.length * width
	}, [tasks, width]);
```
Let's make it more readable

File: apps/mobile/src/libs/helpers/add-notification-on-tasks-expired/add-notifications-on-tasks-expired.helper.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+					],
+					channelId,
+				},
+				body: categoryName,
+				id: id.toString(),
+				title: "You have a task expired",
+			},
+			trigger,
+		);
+	}
```
Comment (fshabanov) [inline apps/mobile/src/libs/helpers/add-notification-on-tasks-expired/add-notifications-on-tasks-expired.helper.ts:49]: Create a separate package for that

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

**Pull request merged**